pkg/controller/clone/host-clone.go

```diff
@@ -209,6 +214,67 @@ func (p *HostClonePhase) createClaim(ctx context.Context) (*corev1.PersistentVol
 	return claim, nil
 }
 
+// The purpose of this check is to make sure the target PVC to have sufficient space
+// before the cloning actually happens.
+func (p *HostClonePhase) MakeSureTargetPVCHasSufficientSpace(ctx context.Context, targetPvc *corev1.PersistentVolumeClaim) error {
+	sourcePvc := &corev1.PersistentVolumeClaim{}
+	sourcePvcKey := client.ObjectKey{Namespace: p.Namespace, Name: p.SourceName}
+
+	if err := p.Client.Get(ctx, sourcePvcKey, sourcePvc); err != nil {
+		return err
+	}
+	if targetPvc.Spec.Resources.Requests != nil {
+		unInflatedSourceSize := sourcePvc.Spec.Resources.Requests[corev1.ResourceStorage]
+		inflate := false
+		sourceVolumeMode := cc.GetVolumeMode(sourcePvc)
+		targetVolumeMode := cc.GetVolumeMode(targetPvc)
+		var err error
+
+		// For filesystem source PVCs, we need to get the original DV size to account for
+		if sourceVolumeMode == corev1.PersistentVolumeFilesystem {
+			original := cc.GetHostCloneOriginalSourceDVSize(ctx, p.Client, sourcePvc)
+			if !original.IsZero() {
+				unInflatedSourceSize = original
+			}
+		}
+
+		targetSize := targetPvc.Spec.Resources.Requests[corev1.ResourceStorage]
+
+		if unInflatedSourceSize.Cmp(targetSize) > 0 {
+			targetSize = unInflatedSourceSize
+			// if both target and source have same volume mode, take a original size if target is smaller
+			// no need to inflate
+
+			// if filesystem to block, inflate original size if block size is smaller
+			if sourceVolumeMode == corev1.PersistentVolumeFilesystem && targetVolumeMode == corev1.PersistentVolumeBlock {
+				inflate = true
+			}
+		}
+
+		// if block to filesystem, inflate the original size if target is smaller
+		// need to take the filesystem overhead into account when comparing sizes
+		if sourceVolumeMode == corev1.PersistentVolumeBlock && targetVolumeMode == corev1.PersistentVolumeFilesystem {
```

Comment:
At this point the uninflated size is one of two options:
1. The source is block, and thus it is just the source size.
2. The source is filesystem and we looked up the original size of the DV.

Either way, the source is now correct, we only need to inflate it if the target is filesystem. So no need for `sourceVolumeMode == corev1.PersistentVolumeBlock`